### PR TITLE
Blog post about our engineering culture

### PR DESCRIPTION
**Pull request by phereford**

### CODE REVIEW

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
+---
+layout: post
+title:  "Our set of engineering culture and values"
+date:   2016-09-30 14:12:09 -0400
+categories: engineering culture
+---
+
+Every engineering group is different. Every engineering manager values different
+aspects of its team. This list was put together by the engineering team at
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: `its` should be his, her or their (if you want to be truly inclusive for gender non-conformists)

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+_Forward Financing_.  
+
+## Engineer's Flow  
+There is this engineering management theory called "Flow". It's a state any
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: There is an engineering management theory called "Flow" -- it's any state that is void of interruption and disruption for an engineer.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+engineer achieves to get into that is void of interruption and disruption. This
+is when engineers do their best work, can tackle the hardest problems, and why
+you'll often find our room either full of energy and boisterous or very very
+quiet to the point you can hear a pin drop. We do many things to help get
+engineers into Flow, some rules we follow:
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Flow. Some

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+## Engineer's Flow  
+There is this engineering management theory called "Flow". It's a state any
+engineer achieves to get into that is void of interruption and disruption. This
+is when engineers do their best work, can tackle the hardest problems, and why
+you'll often find our room either full of energy and boisterous or very very
+quiet to the point you can hear a pin drop. We do many things to help get
+engineers into Flow, some rules we follow:  
+1) Do not disturb if headset is on.
+2) Turn off slack notifications when working on larger/harder problems.
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Slack, capitalize

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+3) Flexible work from home policy to reduce visual noise and disruption from
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: work-from-home

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+outside parties.
+
+Here are some articles that speak specifically about flow or topics close in
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Flow in the context of this blog post should be capitalized

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+- http://www.paulgraham.com/makersschedule.html
+- http://firstround.com/review/track-and-facilitate-your-engineers-flow-states-in-this-simple-way/
+- https://hackernoon.com/its-a-wonder-some-teams-get-anything-done-64104f59bf5e
+  
+## Scrum  
+Scrum is an Agile methodology that help teams get things done in as productive
+a manner as possible. It allows the business to track project progress
+transparently and get an idea of throughput based on estimation. Scrum allows
+the product team (Product Managers and Engineer) to be able to set expectations
+that stakeholders understand and abide by. While none of us believe in burn down
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Line break before "While"

Comment (kelvinma) [reply _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: `believes`, none is singular "not one of us believes"

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+charts, for those people that are far removed from the process, it can act as
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: people who are

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+a simple flag as to a sprint going well, or falling behind. The most important
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: flag that a sprint is going well or falling behind.

No need for that comma.

Comment (kelvinma) [reply _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Line break after 'behind'

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+piece of Scrum is the Retrospective. This is an open forum to discuss what went
+well during the sprint, what failed, and what we can do as a team to address
+the issues that led to the failures. Thus far, we have evolved our process with
+our current team to address the issues that have arisen so far, and it has all
+been for the better. You can find our retrospective notes in Confluence.
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: This blog post is public facing. Rephrase to say that something like "We archive our retrospective notes in Confluence", unless we plan on releasing those notes to the public.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+
+Here are some articles that speak specifically about Scrum and it's importance
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: `its` importance

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+in organizations.
+- https://martinfowler.com/bliki/FlaccidScrum.html
+- https://www.scrumalliance.org/community/articles/2014/november/top-10-reasons-to-adopt-scrum-in-an-organization
+  
+## Blameless Culture  
+Having a blameless culture is paramount for building a sustainable, long lived
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: long-lived

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+engineering culture. No matter the project, bugs are going to exist. Whether
+it is due to miscommunication, misinterpretation, speed, lack of tests, etc,
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: etc.,

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+bugs will always be introduced into new and old systems alike. It is for that
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~It is for that reason that~

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+reason that it is of utmost importance to have a culture of _learning from
+mistakes_ as opposed to vilifying the one individual who made the mistake.
+
+Here are some links that better describe the nuance of a blameless culture and
+it's benefits to a team and it's culture.
+- https://americastestkitchen.github.io/culture/engineering/2015/05/13/the-value-of-postmortems/
+  
+## Technical Debt  
+Every company has technical debt, and it can take many forms. Some are legacy
+projects that engineering teams inherit with it's primary architects long gone.
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: its

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+Other forms are when a deadline is very tight that some short cuts are taken
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Sometimes shortcuts are taken in order to meet deadlines.

A little more concise.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+in order to hit deadlines. In either scenario, technical debt has a deliterious
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: it's spelled deleterious, and i think you can use a better word that people won't have to use a dictionary to look up. Harmful? Negative?

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+effect on health of systems, engineering productivity, and more. This topic,
+which has been discussed in length in every possible forum, is of the utmost
+importance due to it's impact.
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~it's~ its

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+
+Here are some links about what technical debt is, and it's place in
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~it's~ its

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 20]
```diff
@@ -0,0 +1,179 @@
...
+organizations.
+- https://martinfowler.com/bliki/TechnicalDebt.html
+- https://medium.com/modern-agile/who-manages-technical-debt-levels-7e634b0a5974?source=linkShare-fd418bfc22ba-1506544066
+  
+## Code Review  
+Code review is very important in every organization. It acts as a way to
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Three "it acts" in a row. Maybe give your penchant for the oxford comma some love? 😎

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 21]
```diff
@@ -0,0 +1,179 @@
...
+  
+## Continuous Delivery  
+There are a lot of pros and cons around having a Continuous Delivery setup. We
+have most of our applications (all but our application that handles money) on
+continuous delivery. This methodology removes the ceremony around deployment.
+Removing the deployment ceremony reduces engineer frustration, increases
+engineer productivity, and so much more. From a business standpoint, features
+get released faster and quicker.
+
+Here are some articles about what continuous delivery is and it's benefits to an
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~it's~ its

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 22]
```diff
@@ -0,0 +1,179 @@
...
+organization.
+- https://martinfowler.com/bliki/ContinuousDelivery.html
+
+## Pair Programming  
+It's difficult to concisely speak about the importance of pair programming.
+Beyond the immediate impact it provides engineers, it also has long lasting
+effects that can't be replaced with other frameworks. It helps form bonds of
+mentorship and coaching as to improving the engineer that requested the pair. It
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: toward improving the engineer who requested...

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 23]
```diff
@@ -0,0 +1,179 @@
...
+and help each other through challenges.
+
+While there are plenty of pro and con articles around the internet about
+pairing, here are some about the positive benefits of it and why we use it
+at Forward Financing:
+- https://raygun.com/blog/how-good-is-pair-programming-really/
+  
+## Testing  
+Testing is another important vehicle that helps reduce technical debt from
+being introduced into new systems and helps vet that requirements are met. Tests
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~helps vet~ ensures

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 24]
```diff
@@ -0,0 +1,179 @@
...
+are a way an engineer can programatically understand if it's functionality is
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~it's~ its

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 25]
```diff
@@ -0,0 +1,179 @@
...
+Here are some articles about why testing is important and why companies decide
+to invest in testing.
+- https://martinfowler.com/bliki/UnitTest.html
+  
+## Continuous Learning  
+Continuous learning is nothing new to businesses, but has made a recent
+resurgence in business culture. This framework allows companies to invest in
+employees by giving them new skills, reinforcing old skills with new twists, and
+more. This has so many positive effects on an organization that it is difficult
+to enumerate them all. Two that I tend to focus on are:
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Two on which I tend to focus:

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 26]
```diff
@@ -0,0 +1,179 @@
...
+1) Employee retention
+2) Employee growth.
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: no period

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 27]
```diff
@@ -0,0 +1,179 @@
...
+
+Here are some links around the internet that describe what continuous learning
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~around the internet~ it's a link. we know it's on the internet.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 28]
```diff
@@ -0,0 +1,179 @@
...
+is and why it is important.
+- https://www.td.org/Publications/Magazines/TD/TD-Archive/2017/02/Webex-10-Ways-to-Build-a-Culture-of-Continuous-Learning
+  
+## Management  
+Managing a team of engineers is often likened to herding cats. It is a tall task
+that most people fumble through. Being the liason between senior management and
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: liaison

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 29]
```diff
@@ -0,0 +1,179 @@
...
+engineers also has it's own set of challenges. The two most important pieces
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: ~it's~ its

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 30]
```diff
@@ -0,0 +1,179 @@
...
+that any manager can do for engineers are:
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: engineers: ~are:~
the verb negates the need for the colon, but having a bullet point list makes it easier to read.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 31]
```diff
@@ -0,0 +1,179 @@
...
+1) Guard rigorously against context switching.
+2) Facilitate growing trust throughout the team.
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: don't need the periods here

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 32]
```diff
@@ -0,0 +1,179 @@
...
+
+Beyond those two, there is so much that happens to keep engineers happy,
+motivated, and engaged. Here are some links from individuals that I respect in
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Here are some articles written by people who I respect...

You respect the people, not the links.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 33]
```diff
@@ -0,0 +1,179 @@
...
+the industry that have gone on to build some amazing engineering cultures:
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: industry who have

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 34]
```diff
@@ -0,0 +1,179 @@
...
+- http://firstround.com/review/three-powerful-conversations-managers-must-have-to-develop-their-people/
+- https://hackernoon.com/40-hours-is-enough-86d7166911ea
+- https://m.signalvnoise.com/nobody-hits-the-ground-running-ae4d8fd0e7de
+- https://m.signalvnoise.com/software-has-bugs-this-is-normal-f64761a262ca
+- https://medium.com/startup-patterns/the-dangers-of-measuring-performance-d21cd61426df
+- https://medium.com/javascript-scene/getting-to-10x-results-what-any-developer-can-learn-from-the-best-54b6c296a5ef
+- https://home.opsolutely.com/the-hidden-reason-you-cant-scale-your-engineering-team/
+  
+## Product Development  
+The process around how products are developed are important, but not the end
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: developed is important <- referring to process

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 35]
```diff
@@ -0,0 +1,179 @@
...
+all be all. It is important to have structures and processes in place so others
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: end-all and be-all

Comment (kelvinma) [reply _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: processes in place to foster accountability, or something like that.

"others know how to keep others" reads awkward

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 36]
```diff
@@ -0,0 +1,179 @@
...
+know how to keep others accountable. These structures help prevent friction
+between other groups as they work together through a common goal. With that
+said, the product will always beat the process. Until something hits the hands
+of a customer, nothing matters from the point of view of the customer.
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Nothing matters until the product is in the hands of the customer.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 37]
```diff
@@ -0,0 +1,179 @@
...
+between other groups as they work together through a common goal. With that
+said, the product will always beat the process. Until something hits the hands
+of a customer, nothing matters from the point of view of the customer. 
+
+Here are some links:
+- https://hackernoon.com/antifragile-product-development-39b8d5866b77
+- https://medium.com/pminsider/the-product-beats-the-process-28028151c485
+  
+## Transparency  
+It's no secret about the positive effects of transparency. Being transparent
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: The positive effects of transparency is no secret. 

Reads a bit awkward.

**Comment by jks8787** [comment]
https://code.likeagirl.io/the-power-of-giving-good-feedback-78ab2f4c2b72
https://code.likeagirl.io/how-to-show-all-important-humility-in-leadership-without-being-a-doormat-a90433614ce6
https://code.likeagirl.io/6-solutions-to-my-recruitment-problem-but-its-still-a-problem-d299b6a52c6c
https://medium.com/the-year-of-the-looking-glass/how-do-you-set-metrics-59f78fea7e44
https://m.signalvnoise.com/the-3-types-of-bullshit-feedback-and-what-to-do-about-them-5de483530c3f

These are links that may be grouped with some of the others, these ones are from author's with differnt profiles that the ones you pulled from for this post. I think after looking at all those links, and thinking on it, I felt like we were getting a lot from a certain set of voices. might be nice to mix it up

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md (hunk 5 above)
Comment (kelvinma) [reply _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: Remove the periods

**Comment by phereford** [comment]
@jks8787 Added to the management section. IF you think I should create a new section regarding this around Feedback/Growth let me know.

File: _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md [hunk 38]
```diff
@@ -0,0 +1,189 @@
...
+## Management  
+Managing a team of engineers is often likened to herding cats. It is a tall task
+that most people fumble through. Being the liaison between senior management and
+engineers also has its own set of challenges. The two most important pieces
+that any manager can do for engineers:
+1) Guard rigorously against context switching
+2) Facilitate growing trust throughout the team
+
+Beyond those two, there is so much that happens to keep engineers happy,
+motivated, and engaged. Here are some articles written by people who I respect in
```
Comment (kelvinma) [inline _posts/2017-09-28-our-engineering-culture-methodologies-and-philosophies.md]: whom

**Pull request merged**